spec/views/shared/activities/activity_spec.rb

```diff
@@ -112,46 +116,91 @@
+              expect(rendered).not_to have_content(t("activerecord.attributes.activity.oda_eligibility"))
+              expect(rendered).not_to have_content(t("activerecord.attributes.activity.oda_eligibility_lead"))
+            end
           end
         end
 
-        context "when the activity is non-ODA" do
-          let(:activity) { build(factory_name, :ispf_funded, ispf_theme: 1, ispf_partner_countries: ["IN"], is_oda: false) }
-
-          it "doesn't show fields irrelevant to ISPF non-ODA #{level}s" do
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.objectives"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.benefitting_countries"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.gdi"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.collaboration_type"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.sustainable_development_goals"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.aid_type"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.fstc_applies"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.policy_marker_gender"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.policy_marker_climate_change_adaptation"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.policy_marker_climate_change_mitigation"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.policy_marker_biodiversity"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.policy_marker_desertification"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.policy_marker_disability"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.policy_marker_disaster_risk_reduction"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.covid19_related"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.channel_of_delivery_code"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.oda_eligibility"))
-            expect(rendered).not_to have_content(t("activerecord.attributes.activity.oda_eligibility_lead"))
+        context "and it has a linked activity" do
+          let(:activity) {
+            create(factory_name, :ispf_funded,
+              extending_organisation: user.organisation,
+              organisation: user.organisation,
+              ispf_theme: 1,
+              ispf_partner_countries: ["IN"],
+              linked_activity: create(:third_party_project_activity, :ispf_funded))
```

Comment:
Given we're using `factory_name` here to cover both projects and third-party projects, does the `linked_activity` also need to use the factory name - otherwise we might be linking activities at different levels?

Wonder if this is getting missed because we don't have model validations?